packages/flutter/lib/src/widgets/actions.dart

```diff
@@ -508,11 +524,20 @@ class Actions extends StatefulWidget {
   /// that if the actions change, the context will be rebuilt and find the
   /// updated action.
   static Action<T>? find<T extends Intent>(BuildContext context, {bool nullOk = false}) {
-    Action<T>? action;
+    return findByType(context, T, nullOk: nullOk) as Action<T>?;
+  }
+
+  /// Finds the [Action] bound to the given intent type `type` in the given `context`.
+  ///
+  /// Creates a dependency on the [Actions] widget that maps the bound action so
+  /// that if the actions change, the context will be rebuilt and find the
+  /// updated action.
+  static Action<Intent>? findByType(BuildContext context, Type type, {bool nullOk = false}) {
```

Comment:
Looks like you figured out why below.

Reply:
Could we remove `findByType` and instead change the signature of `find` to:

```dart
  static Action<T>? find<T extends Intent>(BuildContext context, {bool nullOk = false, T? intent})
```

The method body would ignore the `intent` argument. It's only there to infer `T` if you you only have an Intent instance available, but don't know its type statically?

I believe you can then call `find(context, intent: matchedIntent)` from wherever you need to call it and it would correctly infer `T`. I think?

If someone knows the type statically, but doesn't have an intent instance, they can still use this API since `intent` is optional.

Reply:
Could we remove `findByType` and instead change the signature of `find` to:

```dart
  static Action<T>? find<T extends Intent>(BuildContext context, {bool nullOk = false, T? intent})
```

The method body would ignore the `intent` argument. It's only there to infer `T` if you you only have an Intent instance available, but don't know its type statically?

I believe you can then call `find(context, intent: matchedIntent)` from wherever you need to call it and it would correctly infer `T`. I think?

If someone knows the type statically, but doesn't have an intent instance, they can still use this API since `intent` is optional.

Reply:
OK, I'll give that a try.

Reply:
It works!